File: java_solutions/1732.find-the-highest-altitude.java
/*
 * @lc app=leetcode id=1732 lang=java
 *
 * [1732] Find the Highest Altitude
 */

class Solution {
    public int largestAltitude(int[] gain) {
        return IntStream.range(0, gain.length + 1)
            .map(i -> Arrays.stream(gain).limit(i).sum())
            .max()
            .orElse(0);
    }
}

File: kotlin_solutions_1/1732.find-the-highest-altitude.kt
/*
 * @lc app=leetcode id=1732 lang=kotlin
 *
 * [1732] Find the Highest Altitude
 */

class Solution { fun largestAltitude(gain: IntArray): Int = gain.fold(0 to 0) { (max, curr), g -> (curr + g).let { maxOf(max, it) to it } }.first }